### DISCUSSION TITLE
Default Strategy wont be set for new installations

### DISCUSSION THREAD

**Discussion by wonderwilly** [rustdesk/rustdesk#7074] [Q&A]
Hi there.
Running RustDesk Server Pro 1.1.14 and enabled and configured Default Strategy, e.g. disabled TCP tunneling and filetransfer.
But when installing new Windows clients via
`\\...\rustdesk-licensed-xxxxxxxxxx.exe --silent-install`
they wont take over the default configuration and settings like TCP tunneling and filetransfer are still enabled.
Also tried:
`"C:\Program Files\RustDesk\rustdesk.exe" --config "xxxxxxx"`
Do that strategies not work as I think and if not, what can I do to push/force same config to each new Windows client?
Thanks a lot.

**Comment by 21pages** [comment]
I can reproduce it with `\\...\rustdesk-licensed-xxxxxxxxxx.exe --silent-install`, It seems that the licensed config will not work after installation, because the file name becomes "rustdesk.exe",  licensed config is used for portable version.
If installed this way, it will still use the public server, when I changed the server configuration manually, the strategy worked.
If using `rustdesk-licensed-xxxxxxxxxx.exe` in portable way, the strategy will work without any configuration.

Please check whether it is using public server.
Public server
![1707271318998](https://github.com/rustdesk/rustdesk/assets/14891774/21052386-0815-428f-9deb-c00fcb84863f)

Self host server
![1707271061191](https://github.com/rustdesk/rustdesk/assets/14891774/bfc1bed5-91f7-4ad3-acfa-dfd34d304b44)

**Reply by wonderwilly** [reply]
Hello thanks for quick reply. Unfortunately I can not confirm: When installing via `\\...\rustdesk-licensed-xxxxxxxxxx.exe --silent-install`, the status line in the client says "Ready" and the client is shown in servers device list with green dot, status = "normal" and Strategy = "-" (stand for Default Strategy I think), so it definitely uses the own server. Do you no whether there are other ways to force/deploy the prefered configs, e.g. via registry?

**Reply by rustdesk** [reply]
In devices page, can you see this device? on the device column, can you see the device name and user name like this `<username>@<devicenmae>`?

**Reply by 21pages** [reply]
1. Could you confirm if the strategy is enabled?
enabled:
![chrome_HbHlh1JOJY](https://github.com/rustdesk/rustdesk/assets/14891774/bc9a4ac9-0672-49fb-ac66-4fe8f4eaa8f1)
disabled:
![chrome_GMB67G6cPL](https://github.com/rustdesk/rustdesk/assets/14891774/90a3988e-59af-4ac4-b239-b3d24c33a0eb)
2. Check if there is a option `strategy_timestamp` in file `C:\Windows\ServiceProfiles\LocalService\AppData\Roaming\RustDesk\config\RustDesk_local.toml`, if there is, remember the value.
3. Do some changes to the default strategy and save, wait about 1 minute, check whether the value of option `strategy_timestamp` is changed, and whether the strategy has worked.

**Reply by wonderwilly** [reply]
Hello again.
@rustdesk   yes the client is shown in the device list with IP and user.
@21pages   the default strategy realy was disabled :-/ enabled it and rolled out client again to test computer - settings now are correct!! Will try with some more computers but I think that was ist. Thanks for quick and helpful replies!! I will get back to you after some more testings.

**Reply by 21pages** [reply]
Thanks, we need to make the enabled/disabled status more clear.

**Reply by rustdesk** [reply]
Add `(disabled)` right of the name, change the text color to grayed color.

**Reply by wonderwilly** [reply]
I think it works now, thanks again. One more question: After silent installing via `\\...\rustdesk-licensed-xxxxxxxxxx.exe --silent-install` the client window appears - is it possible to avoid the client from automatically appearing?

**Reply by 21pages** [reply]
I can't reproduce it with 1.2.3, 1.2.4

**Reply by wonderwilly** [reply]
OK you're right. After (silent) installing the client does not appear. Only when making Rustdesk running as a serive via `\\...\rustdesk-licensed-xxxxxxxxxx.exe --install-service` it appears. Sorry, scripted both commands (--silent-install and --install-service) one after another and did not realize that the second (service) command is responsible for client appearing. Is it possible to avoid appearing?

**Reply by 21pages** [reply]
"--install-service" is not necessary if the service was not previously set to stop.

**Reply by wonderwilly** [reply]
Ah OK `silent-install` will make it run as service, too? Will test it....

**Reply by wonderwilly** [reply]
Nice! Thanks!
For MAC users there is no possibility to prepare the portable client like for windows with `rustdesk-licensed-xxxxxxxxxx.exe`, right?

**Reply by 21pages** [reply]
Right

**Comment by wonderwilly** [comment]
Hello again. BTW: Should I create a new post for new questions not relating the Default Strategy or could I use this thread further? I just do it - please let me know whether to psot new question :-)
One test user downloaded both - the default Win client from rustdesk.com and our individual client rustdesk-licensed-xxxxxxxxxx.exe but says that his home PC does not do anything when trying to run the clients. Tried with admin privileges, tried to disable the local antivir, nothing. The client does not start up and in taskmanager there is nothing to see. Any ideas?

**Reply by 21pages** [reply]
If it's the latest nightly build, pleaset test 1.2.3 https://github.com/rustdesk/rustdesk/releases/tag/1.2.3, if it doesn't work, try sciter version, and please tell the os version, cpu, gpu.

**Reply by wonderwilly** [reply]
It's no nighley version, just the downloadable version from rustdesk.com start page: Rustdesk 1.2.3+39.
OS: Windows 10 Pro, 10.0.19045 Build 19045
CPU: Core i7-7700
GPU: NVIDIA GeForce GTX 1060 6GB

**Reply by 21pages** [reply]
try https://github.com/rustdesk/rustdesk/releases/download/1.2.3/rustdesk-1.2.3-x86-sciter.exe

**Reply by wonderwilly** [reply]
ok, thanks, will tell him and let him test it. what is sciter, kind of beta? and could we make it individual, too? or does he need to manually fill in the cleints settings for ID-server, API-server, key and so on?

**Reply by 21pages** [reply]
legacy ui, needn't if not installed.

**Reply by wonderwilly** [reply]
OK without our specific server information he can not connect, it says remote client is offline, but can not set our server infos because no menu button in the legayc client? Need to install it (wouldnt want to tell him need to install...).

**Reply by 21pages** [reply]
![rustdesk_Ar8tcBMMlL](https://github.com/rustdesk/rustdesk/assets/14891774/c578e0b0-cbce-4572-a6c0-97178ae3eec4)
